### PR TITLE
feat: add chat integration

### PR DESCRIPTION
**Pull request by ShaikMoAbdullah**
- [x] Added chat integration
- [x] Fixed issues with respect to CORS error by allowing all origins from the backend (temporary fix). 
Screenshot from 2024-10-25 16-46-29
Screenshot from 2024-10-25 16-46-46
Screenshot from 2024-10-25 16-47-26
Screenshot from 2024-10-25 16-48-33

### CODE REVIEW

File: frontend/src/screens/Chat.tsx [hunk 1]
```diff
@@ -99,72 +82,118 @@ export function ChatBot({ pastMessages = [] }: ChatBotProps) {
...
+      setInput("");
+      setError("");
+      setIsFetching(true);
+      setConversationStarted(true);
+
+      try {
+        let collectedPayload = "";
+        let updatedSessionId = sessionId;
+
+        const response = await fetch("http://127.0.0.1:8000/chat", {
```
Comment (SoumyaRanjanPatnaik) [inline frontend/src/screens/Chat.tsx:119]: Probably a good idea to keep the urls in a .env file. 

Link for reference: https://vite.dev/guide/env-and-mode#env-files

File: backend/fastapi-sse/server-sse.py [hunk 2]
```diff
@@ -59,36 +70,28 @@ def generate_sql_query_responses(
...
             session_id = str(uuid.uuid4())
             logger.info(f"Generated new session_id: {session_id}")
 
-        # If no type is provided, set it to default option of "report"
         if not type:
-            type = "report" 
+            type = "report"
 
-        # Returning the StreamingResponse with the proper media type for SSE
         logger.info(f"Started streaming SQL responses for query: {query}")
```
Comment (SoumyaRanjanPatnaik) [inline backend/fastapi-sse/server-sse.py:95]: Why do we need both GET and POST? I guess just using POST would do. Also, we don't need to manually extract data from the request body. FastAPI should do this automatically for us. (reference: https://fastapi.tiangolo.com/tutorial/body/)

Can we also bring back the comments?😅

File: frontend/src/screens/Chat.tsx [hunk 3]
```diff
@@ -1,89 +1,72 @@
...
-  return (
-    <Container
-      bg="gray.900"
-      centerContent
-      alignItems="center"
-      justifyContent="center"
-      maxW="100vw"
-      h="100%"
-    >
+  const botGreeting = useMemo(
```
Comment (SoumyaRanjanPatnaik) [inline frontend/src/screens/Chat.tsx:68]: Can be a seperate component I guess?

File: frontend/src/screens/Chat.tsx (hunk 1 above)
Comment (SoumyaRanjanPatnaik) [reply frontend/src/screens/Chat.tsx:119]: Might be a good idea to move the API stuff to another function (outside this component). Would clean up the component a little bit. Your call though.

File: backend/fastapi-sse/server-sse.py [hunk 4]
```diff
@@ -16,6 +18,15 @@
...
 
+# Configure CORS
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],  # Allows all origins
+    allow_credentials=True,
+    allow_methods=["*"],  # Allows all methods (GET, POST, etc.)
+    allow_headers=["*"],  # Allows all headers
+)
+
```
Comment (SoumyaRanjanPatnaik) [inline backend/fastapi-sse/server-sse.py:29]: Might be a good idea to do get the frontend URL from the .env file
```suggestion
from dotenv import load_dotenv
load_dotenv()

# Configure CORS
allowed_origin = os.environ.get("CORS_ORIGINS", "").split(" ")
app.add_middleware(
    CORSMiddleware,
    allow_origins=allowed_origin,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

Inside the `.env` file:
```bash
# Frontend URL
CORS_ORIGINS=http://localhost:3000 
```

**Review by SoumyaRanjanPatnaik — CHANGES_REQUESTED** [review]

File: backend/fastapi-sse/server-sse.py (hunk 2 above)
Comment (ShaikMoAbdullah) [reply backend/fastapi-sse/server-sse.py:95]: `Also, we don't need to manually extract data from the request body. FastAPI should do this automatically for us.`

@MalayJoshi-Karya Please make this change

Comment (ShaikMoAbdullah) [reply backend/fastapi-sse/server-sse.py:95]: @SoumyaRanjanPatnaik Had a very little idea about this. Much of it was done with the help of ChatGPT :smile:

File: frontend/src/screens/Chat.tsx (hunk 1 above)
Comment (ShaikMoAbdullah) [reply frontend/src/screens/Chat.tsx]: Done

File: backend/fastapi-sse/server-sse.py (hunk 2 above)
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/server-sse.py]: @ShaikMoAbdullah the code which you got from Chatgpt has been already fixed by me in the following repo: #14 , you can resolve this conversation.

File: backend/fastapi-sse/server-sse.py (hunk 4 above)
Comment (MalayJoshi-Karya) [reply backend/fastapi-sse/server-sse.py]: @ShaikMoAbdullah the code which you got from Chatgpt has been already fixed by me in the following repo: #14 , you can resolve this conversation as well.

**Comment by ShaikMoAbdullah** [comment]
@MalayJoshi-Karya Please look into these merge conflicts

**Comment by ShaikMoAbdullah** [comment]
Resolved all the merge conflicts

**Review by SoumyaRanjanPatnaik — APPROVED** [review]
LGTM

**Pull request merged**